### PR TITLE
Parquet Writer: Make column descriptor public on the writer

### PR DESCRIPTION
**Pull request by pier-oliviert**
# Which issue does this PR close?
There's no issue currently, because I just had this issue this morning and decided to make a PR for it. I can open up an issue if it helps.

# Rationale for this change

This is so that it's possible to gather information from the column we're about to write to. That information was already present in the column but buried inside the internal of the GenericColumnWriter.

The getter returns a reference of `ColumnDescPtr`.

# What changes are included in this PR?
The PR is missing tests, I wasn't sure if the test would add any value/safety. I'm more than happy to provide some tests if you think it's important.

# Are there any user-facing changes?
None, I believe.

### CODE REVIEW

File: parquet/src/file/writer.rs [hunk 1]
```diff
@@ -473,9 +479,14 @@ impl<'a> SerializedColumnWriter<'a> {
     /// optional callback to be invoked on [`Self::close`]
     pub fn new(
         inner: ColumnWriter<'a>,
+        descriptor: ColumnDescPtr,
```
Comment (tustvold) [inline parquet/src/file/writer.rs:482]: This is a breaking change

File: parquet/src/file/writer.rs [hunk 2]
```diff
@@ -488,6 +499,11 @@ impl<'a> SerializedColumnWriter<'a> {
         get_typed_column_writer_mut(&mut self.inner)
     }
 
+    /// Returns a clone to a [`ColumnDescPtr`]
+    pub fn descriptor(&self) -> ColumnDescPtr {
```
Comment (tustvold) [inline parquet/src/file/writer.rs:503]: Returning &ColumnDescPtr would be more idiomatic so the caller can clone only if they need to

**Review by tustvold — COMMENTED** [review]
Thank you for the PR, I'm not sure about this in particular the breaking API change. Perhaps you could expand a bit on your use-case?

**Comment by pier-oliviert** [comment]
@tustvold Maybe it's possible to do what I'm trying to do using a different API, but currently, what I'm working on is a way for me to aggregate events that comes in from a different thread, and package data into a parquet file using the SerializedColumnWriter.

What this means, is that I have an in-memory cache that I'm holding, and then constructing the parquet file (row group) when it reaches a certain size.

Since the ColumnDescriptor is not really available, as far as I can tell, I have to basically create an index myself to figure out which column I am referring to when I call `writer.next_column()`.

If I had the possibility to gather some kind of information from the columnWriter, it would help understand the context I'm in and avoid some bookkeeping on my side. 

Does that make sense?

I apologize for the breaking change, I guess I misinterpreted what "User-facing changes" meant.

**Comment by pier-oliviert** [comment]
Now that I'm looking at the problem again this morning, I believe that moving the getter from `SerializedColumnWriter` to [`GenericColumnWriter`](https://github.com/apache/arrow-rs/blob/master/parquet/src/column/writer/mod.rs#L191) would solve my problem.

The struct already has that field and so just adding a getter to return a reference to it would work. I believe this would not introduce a breaking change?

**Comment by tustvold** [comment]
> If I had the possibility to gather some kind of information from the columnWriter, it would help understand the context I'm in and avoid some bookkeeping on my side.

Makes sense :+1:

>  I believe this would remove the breaking change of the API and solve my problem, would that be an acceptable solution?

Sounds good to me

**Review by tustvold — APPROVED** [review]
Thank you for sticking with this :+1:

**Pull request merged**